hackathon_site/hardware/test_api.py

```diff
@@ -136,6 +136,41 @@ def test_category_get_success(self):
 
         self.assertEqual(expected_response, data["results"][0])
 
+    def test_hardware_quantity_success(self):
+        self._login()
+
+        self.category = Category.objects.create(name="Microcontrollers", max_per_team=4)
```

Comment:
Note the best to overwrite the `self.category` that's already defined, could make this `self.category2`. Though since you're not using it outside of this test case, you don't actually need the `self.` (same for `hardware1` and `hardware2` below)

Reply:
@grahamhoyes  So only when we want to use the variable in the other test within the same class, we need to use "self" right? Is that possible to use the variable in the other test in another class?

Reply:
This is the basic process of what happens for every single test method:
- The database is reset
- The setUp() method on the test class is called, which sets up your default data
- The particular test method is called

Anything you create in the setUp() method that you want accessible in other test methods, you should define on `self`. Inside the test functions themselves, nothing else will need (or be able to) access them, so no need to use `self`.

> Is that possible to use the variable in the other test in another class?

In another class entirely, no. If you mean another test function, then also no. Only things defined in the `setUp()` method will be accessible to other functions.

There are a few caveats to all of this, but for the most part, if you're not accessing a variable from another method in the class, no need to use `self`.